rsfbclient-native/src/connection.rs

```diff
@@ -455,4 +455,60 @@ impl FirebirdClient for NativeFbClient {
 
         Ok(Some(cols))
     }
+
+    fn execute2(
+        &mut self,
+        mut db_handle: Self::DbHandle,
+        mut tr_handle: Self::TrHandle,
+        mut stmt_handle: Self::StmtHandle,
+        params: Vec<SqlType>,
+    ) -> Result<Vec<Column>, FbError> {
+        let (out_xsqlda, _) = self
+            .stmt_data_map
+            .get(&stmt_handle)
+            .ok_or_else(|| FbError::from("Tried to fetch a dropped statement"))?;
+
+        let params = Params::new(
+            &mut db_handle,
+            &mut tr_handle,
+            &self.ibase,
+            &mut self.status,
+            &mut stmt_handle,
+            params,
+            &self.charset,
+        )?;
+
+        unsafe {
+            if self.ibase.isc_dsql_execute2()(
+                &mut self.status[0],
+                &mut tr_handle,
+                &mut stmt_handle,
+                1,
+                if let Some(xsqlda) = &params.xsqlda {
+                    &**xsqlda
+                } else {
+                    ptr::null()
+                },
+                &**out_xsqlda,
+            ) != 0
+            {
+                return Err(self.status.as_error(&self.ibase));
+            }
+        }
+
+        // Just to make sure the params are not dropped too soon
+        drop(params);
+
+        let (_, col_buf) = self
```

Comment:
not too important, but any reason not to just do
`(out_xsqlda, col_buf) = ...`
at the beginning?

Reply:
I've tried, but it gave an error for having multiple mutable borrows of self if I'm not mistaken.
However I'm working on a subsequent pr in the branch `borrowed_handles` that fixes this by not using the `statemet_data_map`.

Reply:
are you sure?
I had no problem to compile it with that change. I'm on on rustc 1.46 in case it matters.
If you feel like sharing the error, here or in private to keep noise down, I am curious

Reply:
Nevermind, it was the one at the `rsfbclient-rust` that needed that, I've tried it in this one and it works

Reply:
> not too important, but any reason not to just do
> `(out_xsqlda, col_buf) = ...`
> at the beginning?

I just followed the others methods syntax.

Reply:
Hmm, so we need to change this now or wait for the `borrowed_handles`?

Reply:
I've removed this on `borrowed_handles`, so i don't think it needs to change now, but i can just merge the changes too, no problem.

Reply:
Ok, so we can deal with this in the `borrowed_handles`